packages/shorebird_tools/lib/src/commands/package_command.dart

```diff
@@ -0,0 +1,70 @@
+import 'dart:io';
+
+import 'package:archive/archive_io.dart';
+import 'package:args/command_runner.dart';
+import 'package:mason_logger/mason_logger.dart';
+import 'package:path/path.dart' as p;
+
+/// {@template package_command}
+///
+/// `shorebird_tools package`
+/// A [Command] that packages a generated patch into an archive
+/// {@endtemplate}
+class PackageCommand extends Command<int> {
+  /// {@macro package_command}
+  PackageCommand({
+    required Logger logger,
+  }) : _logger = logger {
+    argParser
+      ..addOption(
+        'patch',
+        abbr: 'p',
+        mandatory: true,
+        help: 'The patch path to package',
+      )
+      ..addOption(
+        'output',
+        abbr: 'o',
+        mandatory: true,
+        help: 'The path to the output archive',
+      );
+  }
+
+  @override
+  String get description => 'Packages a generated patch into an archive';
+
+  @override
+  String get name => 'package';
+
+  final Logger _logger;
+
+  @override
+  Future<int> run() async {
+    final patchFilePath = argResults!['patch'] as String;
+    final outFilePath = argResults!['output'] as String;
+
+    final patchFile = File(patchFilePath);
+    if (!patchFile.existsSync()) {
+      _logger.err('Patch file not found at $patchFilePath');
+      return ExitCode.software.code;
+    }
+
+    final outFile = File(outFilePath);
+
+    final bytes = patchFile.readAsBytesSync();
+    final archiveFile = ArchiveFile(
+      p.basename(patchFilePath),
+      bytes.length,
+      bytes,
+    );
+
+    ZipEncoder()
+      ..startEncode(OutputFileStream(outFile.absolute.path))
+      ..addFile(archiveFile)
+      ..endEncode();
```

Comment:
Maybe doesn't need to be in this PR, but in the CLI, we've generally put these in an isolate so as not to block progress logs etc.